src/DiscoveryServerManager.cpp

```diff
@@ -1522,18 +1524,21 @@ void DiscoveryServerManager::loadPublisher(
     // default topic is the static HelloWorld one
     const char* profile_name = sub->Attribute(DSxmlparser::PROFILE_NAME);
 
-    PublisherAttributes pubatts;
+    PublisherQos pubqos;
```

Comment:
Previous suggestion applies here as well. So it is not needed to create a DomainParticipant only to extract the XML profiles (which does not even exist).